tests/full_node_test.cpp

```diff
@@ -714,12 +714,18 @@ TEST_F(FullNodeTest, sync_five_nodes) {
   context.assert_all_transactions_known();
   context.assert_balances_synced();
 
-  // Prune state_db of one node to remove all but last 10 block state and verify all ballances still match
+  // Prune state_db of one node
   auto prune_node = nodes[nodes.size() - 1];
-  auto last_block = prune_node->getPbftChain()->getPbftChainSize() - 10;
-  prune_node->getFinalChain()->prune(last_block);
-  context.assert_all_transactions_success();
-  context.assert_all_transactions_known();
+  const uint32_t min_blocks_to_prune = 50;
+  // This ensures that we never prune blocks that are over proposal period
+  ASSERT_HAPPENS({100s, 100ms}, [&](auto &ctx) {
+    const auto max_level = prune_node->getDagManager()->getMaxLevel();
+    const auto proposal_period = prune_node->getDB()->getProposalPeriodForDagLevel(max_level);
+    EXPECT_TRUE(proposal_period.has_value());
```

Comment:
Shouldn't this be WAIT_EXPECT_TRUE?

Reply:
No, this should never be false.

Reply:
So maybe better to use ASSERT_TRUE here? Maybe EXPECT is also ok here, but it would be clear with assert that with instantly fail the test

Reply:
Replaced with ASSERT_TRUE